lib/lld-wrapper/src/link.cc

```diff
@@ -0,0 +1,17 @@
+#include <cstdint>
+
+#include <llvm/Support/raw_ostream.h>
+
+#include <lld/Common/Driver.h>
+#include <lld/Common/ErrorHandler.h>
+
+extern "C" {
+
+void wasmer_lld_wrapper_link(const char **filenames, uint32_t count) {
+  llvm::ArrayRef<const char *> files(filenames, count);
+
+  if (!lld::elf::link(files, false, llvm::outs(), llvm::errs()))
```

Comment:
This linking only supports elf I assume.

Can we adapt the kind of linking depending on the provided format (coff in windows, macho in MacOS)?